src/audio/codec_adapter/codec/cadence.c

```diff
@@ -145,17 +145,17 @@ static int cadence_codec_init(struct comp_dev *dev)
 		setup_cfg->data = rballoc(0, SOF_MEM_CAPS_RAM, codec->cfg.size);
 		if (!setup_cfg->data) {
 			comp_err(dev, "cadence_codec_init(): failed to alloc setup config");
-			module_free_memory(dev, cd);
-			return -ENOMEM;
+			ret = -ENOMEM;
+			goto free;
 		}
 
 		/* copy the setup config */
 		setup_cfg->size = codec->cfg.size;
 		ret = memcpy_s(setup_cfg->data, setup_cfg->size, codec->cfg.data, setup_cfg->size);
 		if (ret) {
-			comp_err(dev, "cadence_codec_init(): failed to copy setup config %d", ret);
-			module_free_memory(dev, cd);
-			return ret;
+			comp_err(dev, "cadence_codec_init(): failed to copy setup config %d",
+				 ret);
+			goto free;
```

Comment:
You can keep the comp_err line unchanged. Not sure why you moved ret on the next line. Line limit was increased to 100 chars so it should be fine.